### PR TITLE
Remove unnecessary podfile targets from rn-tester

### PR DESCRIPTION
**Pull request by Saadnajmi**
#### Please select one of the following
- [x] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

This change removes the ReactApple.nuspec file along with some extra targets we had in our rn-tester pod file. It seems these were here back when we packed and published a NuGet from this repo. Nowadays, that logic happens downstream, and none of our CI mentions this NuGet package or uses those targets. Let's just remove them. 

## Changelog

[Internal] [Removed] - Removed unnecessary pod file targets

## Test Plan

CI Passing should be enough. Would also be worth watching downstream where we consume react-native-macos to make sure nothing breaks there.

### CODE REVIEW

File: packages/rn-tester/Podfile [hunk 1]
```diff
@@ -70,23 +70,6 @@ target 'RNTester-macOSIntegrationTests' do
   pod 'React-RCTTest', :path => "./RCTTest"
 end
 
-# [TODO(macOS GH#774): these are special targets used by the internal Microsoft build pipeline
-target 'iosDeviceBuild' do
```
Comment (HeyImChris) [inline packages/rn-tester/Podfile:74]: Are these not necessary so SDX-Platform can build them later when we pull in rn-macos?

Comment (Saadnajmi) [reply packages/rn-tester/Podfile:74]: SDX-Platform has it's own Podfile in `packages/react-native-community/apple`. In effect, SDX-Platform is the one in charge of turning this public NPM package into a NuGet. I wonder if Android can do the same.

That being said, I expect I could be totally wrong so I'm prepared to back this out if that's the case 😅

**Comment by kelset** [comment]
@Saadnajmi maybe worth rebasing + resolving conflicts then get this one in too?

File: packages/rn-tester/Podfile (hunk 1 above)
Comment (HeyImChris) [reply packages/rn-tester/Podfile:84]: Hey did you get more insight into these? I remember we needed them for CI builds at some point but agree they might be dated now.

**Comment by Saadnajmi** [comment]
@kelset I did pull in the latest. I will probably hold off on merging this change into main / 0.66-stable until after we've finished downstream integration of RN-macOS 0.66 into office. I'm taking that approach juuust to make sure I don't accidentally cause any regressions that delay that upgrade. I don't expect this change to affect anything downstream, but just want to be extra careful =). 

I'll close it for now actually.

**Pull request closed without merging**